src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+            // Push accelerometer biases to hardware registers
+            WriteRegister(Register.XA_OFFSET_H, rawData[0]);
+            WriteRegister(Register.XA_OFFSET_L, rawData[1]);
+            WriteRegister(Register.YA_OFFSET_H, rawData[2]);
+            WriteRegister(Register.YA_OFFSET_L, rawData[3]);
+            WriteRegister(Register.ZA_OFFSET_H, rawData[4]);
+            WriteRegister(Register.ZA_OFFSET_L, rawData[5]);
+
+            // Restore the previous modes
+            WriteRegister(Register.USER_CTRL, (byte)(userControls | (byte)UserControls.I2C_MST_EN));
+            i2cMaster = (byte)(i2cMaster & (~(byte)(I2cBussFrequency.Freqency348kHz) | (byte)I2cBussFrequency.Freqency400kHz));
+            WriteRegister(Register.I2C_MST_CTRL, i2cMaster);
+            DelayHelper.DelayMilliseconds(10, false);
+
+            // Setup I2C for the AK8963
+            //WriteRegister(Register.USER_CTRL, (byte)UserControls.I2C_MST_EN);
+            // Speed of 400 kHz
+            //WriteRegister(Register.I2C_MST_CTRL, 0x0D);
+
+            // Finally store the acceleration bias
+            _accelerometerBias = acceBias / AccSensitivity;
+        }
+
+        /// <summary>
+        /// <![CDATA[
+        /// Run a self test and returns the gyroscope and accelerometer vectores
+        /// a. If factory Self-Test values ST_OTP≠0, compare the current Self-Test response (GXST, GYST, GZST, AXST, AYST and AZST) 
+        /// to the factory Self-Test values (ST_OTP) and report Self-Test is passing if all the following criteria are fulfilled: 
+        /// Axis    | Pass criteria 
+        /// X-gyro  | (GXST / GXST_OTP) > 0.5 
+        /// Y-gyro  | (GYST / GYST_OTP) > 0.5 
+        /// Z-gyro  | (GZST / GZST_OTP) > 0.5
+        /// X-Accel |  0.5 < (AXST / AXST_OTP) < 1.5
+        /// Y-Accel | 0.5 < (AYST / AYST_OTP) < 1.5
+        /// Z-Accel | 0.5 < (AZST / AZST_OTP) < 1.5
+        /// b. If factory Self-Test values ST_OTP=0, compare the current Self-Test response (GXST, GYST, GZST, AXST, AYST and AZST) 
+        /// to the ST absolute limits (ST_AL) and report Self-Test is passing if all the  following criteria are fulfilled.  
+        /// Axis   | Pass criteria
+        /// X-gyro | |GXST| ≥ 60dps  
+        /// Y-gyro | |GYST| ≥ 60dps 
+        /// Z-gyro | |GZST| ≥ 60dps 
+        /// X-Accel| 225mgee ≤ |AXST| ≤ 675mgee
+        /// Y-Accel| 225mgee ≤ |AXST| ≤ 675mgee
+        /// Z-Accel| 225mgee ≤ |AXST| ≤ 675mgee
+        /// c. If the Self-Test passes criteria (a) and (b), it’s necessary to check gyro offset values. 
+        /// Report passing Self-Test if the following criteria fulfilled. 
+        /// Axis   | Pass criteria 
+        /// X-gyro | |GXOFFSET| ≤ 20dps
+        /// Y-gyro | |GYOFFSET| ≤ 20dps
+        /// Z-gyro | |GZOFFSET| ≤ 20dps
+        /// ]]>
+        /// </summary>
+        /// <returns>the gyroscope and accelerometer vectors</returns>
+        public Tuple<Vector3, Vector3> RunGyroscopeAccelerometerSelfTest()
```

Comment:
What is the use case of this method? When would consumers want to call it?

Reply:
To test if the device is actually working correctly. It's about testing if the device is in the proper conditions of work or not. You can run it and send back to telemetry the information, do surveillance with that time to time.